### ISSUE TITLE
Test fails with latest go-crypto

### ISSUE THREAD

**Issue by mdosch** [ProtonMail/gopenpgp#195]
Dear maintainer,

I prepared an update of the go-crypto Debian package to go-crypto v0.0.0-20220812175011-7fcef0dbe794 but this breaks gopenpgp.
It's not caused by the Debian packaging as I get the same issue when checking out your main branch and updating go-crypto:

```
martin@schlepptop ~/build/gopenpgp/crypto (git)-[main] % go test
--- FAIL: TestClearPrivateKey (0.00s)
    keyring_test.go:157: 
        	Error Trace:	keyring_test.go:157
        	Error:      	Expected nil, but got: &errors.errorString{s:"gopenpgp: unknown private key"}
        	Test:       	TestClearPrivateKey
panic: interface conversion: interface {} is *eddsa.PrivateKey, not *ed25519.PrivateKey [recovered]
	panic: interface conversion: interface {} is *eddsa.PrivateKey, not *ed25519.PrivateKey

goroutine 31 [running]:
testing.tRunner.func1.2({0x6be380, 0xc00039b590})
	/usr/lib/go-1.18/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	/usr/lib/go-1.18/src/testing/testing.go:1392 +0x39f
panic({0x6be380, 0xc00039b590})
	/usr/lib/go-1.18/src/runtime/panic.go:838 +0x207
github.com/ProtonMail/gopenpgp/v2/crypto.TestClearPrivateKey(0xc000082d00)
	/home/martin/build/gopenpgp/crypto/keyring_test.go:162 +0x47b
testing.tRunner(0xc000082d00, 0x735a50)
	/usr/lib/go-1.18/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
	/usr/lib/go-1.18/src/testing/testing.go:1486 +0x35f
exit status 2
FAIL	github.com/ProtonMail/gopenpgp/v2/crypto	1.990s
```

**Comment by wussler** [comment]
There are a few breaking changes and I'm taking care to update it here: https://github.com/ProtonMail/gopenpgp/pull/194

**Comment by mdosch** [comment]
Ah, didn't see that. Thanks for the info.

**Comment by wussler** [comment]
This should be fixed with v2.4.9
